app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/image_import_workflow.rb

```diff
@@ -20,87 +51,119 @@ def post_execute
 
-    cleanup_git_repository
-
-    queue_signal(:post_execute_poll, 'importing image into PVS', 'running')
-  end
-
-  def post_execute_poll(*args)
-    msg, status = args
     set_status(msg, status)
+    queue_signal(signal, msg, status)
+  end
 
+  def post_execute_poll(*_args)
     ems = ExtManagementSystem.find(options[:ems_id])
 
     msg    = nil
     status = nil
     signal = nil
+    max_retries = 16
 
     ems.with_provider_connection(:service => 'PCloudTasksApi') do |api|
       response = api.pcloud_tasks_get(context[:task_id])
+      raise MiqException, 'HTTP response object is malformed, remote API error' if response&.status.nil?
 
       case response.status
       when 'capturing', 'downloading', 'creating', 'deleting', 'compressing', 'loading', 'started', 'uploading'
+        context[:retry] = 0
+        msg = "importing image into PVS, current state is: '#{response.status}'"
         signal = :post_execute_poll
-        msg = 'importing image into PVS, current state is: ' + response.status
-        status = 'running'
+        status = 'ok'
       when 'completed'
+        msg = 'importing image into image registry has completed'
         signal = :finish
-        msg = 'importing image into PVS has completed'
         status = 'ok'
       when 'failed'
-        signal = :error
-        msg = 'importing image into PVS has failed'
+        msg = "importing image into image registry has failed with an error: '#{trunc_err_msg(response.status)}'"
+        signal = :abort
+        status = 'error'
+        _log.error(response.status)
+      else
+        raise MiqException, "unexpected status - '#{response.status}'"
+      end
+    rescue => e
+      messsage = if e.kind_of?(IbmCloudPower::ApiError)
+                   suppress(StandardError) do
+                     JSON.parse(e.response_body)["description"] if e.response_body
+                   end
+                 end
+
+      messsage ||= e.message
+
+      retr = context[:retry].to_i
+      if retr < max_retries
+        msg = "failed to retrieve status for a cloud-task: '#{trunc_err_msg(messsage)}'; try number: #{retr + 1} of #{max_retries}"
+        signal = :post_execute_poll
         status = 'error'
+        context[:retry] = retr + 1
+        update!(:context => context)
+        _log.error("failed to retrieve status for a cloud-task: '#{message}'")
       else
-        signal = :error
-        msg = 'PVS API has responded with a non-standard status for a cloud-task: ' + response.status
+        msg = "unable to retrieve cloud-task status after #{max_retries} consecutive tries, see server logs for more details"
+        signal = :abort
         status = 'error'
-        _log.warn(msg)
       end
     end
 
+    set_status(msg, status)
     queue_signal(signal, msg, status)
   end
 
   def post_poll_cleanup
     ems = ExtManagementSystem.find(options[:ems_id])
-    ems.remove_import_auth(options[:import_creds_id])
+    ems&.remove_import_auth(options[:import_creds_id])
```

Comment:
`.find(id)` will raise a `RecordNotFound` exception if the record cannot be found so you don't have to worry about `nil` being returned.  That's if you use `.find_by(:id => id)`